backend/api/app/presentation/user/routes.py

```diff
@@ -87,4 +93,144 @@ def verify_ssid():
+@require_auth
+def logout_user():
+    json_data = request.get_json()
+
+    ssid = g.get("auth_token")
+    ip_address = request.remote_addr
+
+    command = LogoutUserCommand(
+        ssid=ssid,
+        ip_address=ip_address
+    )
+
+    mediator = current_app.config.get('mediator')
+    try:
+            result = mediator.send(command)
+            return jsonify(result), result["status"]
+    except Exception as e:
+            return jsonify({"message": "There was an error during logout.", "error": str(e)}), 500
+        
+@user_bp.route('/general-information', methods=['GET'])
+@require_auth
+def get_general_info():
+    ssid = g.get('auth_token')
+    ip_address = request.remote_addr
+
+    query = GetGeneralInfoQuery(ssid=ssid, ip_address=ip_address)
+    mediator = current_app.config.get('mediator')
+
+    try:
+        result = mediator.send(query)
+        if "user" in result:
+            return jsonify({
+                "message": "User information retrieved successfully",
+                "data": result["user"],
+                "status": 200
+            }), 200
+        else:
+            return jsonify({
+                "message": "User not found",
+                "status": 404
+            }), 404
+    except Exception as e:
+        return jsonify({"message": "An unexpected error occurred.", "status": 500}), 500
+    
+@user_bp.route('/change-password', methods=['PATCH'])
+@require_auth
+def change_password():
+    json_data = request.get_json()
+
+    if not json_data:
+        return jsonify({"message": "No data provided.", "status": 400}), 400
+
+    required_fields = ["old_password", "new_password"]
+    missing_fields = [field for field in required_fields if field not in json_data]
+    if missing_fields:
+        return jsonify({
+            "message": f"Missing required fields: {', '.join(missing_fields)}",
+            "status": 400
+        }), 400
+
+    ssid = g.get('auth_token')
+
+    command = ChangePasswordCommand(
+        ssid=ssid,
+        ip_address=request.remote_addr,
+        data=json_data
+    )
+
+    mediator = current_app.config.get('mediator')
+    try:
+        result = mediator.send(command)
+        return jsonify(result), result["status"]
+    except Exception as e:
+        return jsonify({"message": "There was an error during logout.", "error": str(e)}), 500
+    
+@user_bp.route('/delete-account', methods=['POST'])
+@require_auth
+def delete_account():
+    json_data = request.get_json()
+    if not json_data:
+        return jsonify({"message": "No data provided."}), 400
+    
+    password = json_data.get('password')
+    ssid = g.get("auth_token")
+    
+    if not password:
+        return jsonify({"message": "Password is required."}), 400
+
+    try:
+        ses_id = uuid.UUID(ssid)
+        session = SessionRepository.get_active_by_id(ses_id, request.remote_addr)
+        if not session:
+            return jsonify({"message": "Session not found"}), 404
```

Comment:
This should be check in the handler, not directly in the controller.
Also, if there is no session that is 401, 'Not authenticated'.
Again also, when you take ssid from g.get('auth_token'), that means that that ssid token is correct, otherwise the request would've been terminated because user is not logged in.